src/solc_0.8/asset/AssetSignedAuctionAuthV2.sol

```diff
@@ -0,0 +1,520 @@
+//SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.2;
+
+import "@openzeppelin/contracts-0.8/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-0.8/utils/Address.sol";
+import "../common/Libraries/SigUtil.sol";
+import "../common/Libraries/PriceUtil.sol";
+import "../asset/AssetV2.sol";
+import "../common/Base/TheSandbox712.sol";
+import "../common/BaseWithStorage/MetaTransactionReceiver.sol";
+import "../common/interfaces/ERC1271.sol";
+import "../common/interfaces/ERC1271Constants.sol";
+import "../common/interfaces/ERC1654.sol";
+import "../common/interfaces/ERC1654Constants.sol";
+import "../common/interfaces/IAuthValidator.sol";
+
+contract AssetSignedAuctionAuthV2 is ERC1654Constants, ERC1271Constants, TheSandbox712, MetaTransactionReceiver {
+    struct ClaimSellerOfferRequest {
+        address buyer;
+        address payable seller;
+        address token;
+        uint256[] purchase;
+        uint256[] auctionData;
+        uint256[] ids;
+        uint256[] amounts;
+        bytes signature;
+        bytes backendSignature;
+    }
+
+    enum SignatureType {DIRECT, EIP1654, EIP1271}
+
+    bytes32 public constant AUCTION_TYPEHASH =
+        keccak256(
+            "Auction(address from,address token,uint256 offerId,uint256 startingPrice,uint256 endingPrice,uint256 startedAt,uint256 duration,uint256 packs,bytes ids,bytes amounts)"
+        );
+
+    event OfferClaimed(
+        address indexed seller,
+        address indexed buyer,
+        uint256 indexed offerId,
+        uint256 amount,
+        uint256 pricePaid,
+        uint256 feePaid
+    );
+    event OfferCancelled(address indexed seller, uint256 indexed offerId);
+
+    uint256 public constant MAX_UINT256 = 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff;
+
+    // Stack too deep, grouping parameters
+    // AuctionData:
+    uint256 public constant AuctionData_OfferId = 0;
+    uint256 public constant AuctionData_StartingPrice = 1;
+    uint256 public constant AuctionData_EndingPrice = 2;
+    uint256 public constant AuctionData_StartedAt = 3;
+    uint256 public constant AuctionData_Duration = 4;
+    uint256 public constant AuctionData_Packs = 5;
+
+    mapping(address => mapping(uint256 => uint256)) public claimed;
+
+    IAuthValidator internal _authValidator;
+    AssetV2 public _asset;
+    uint256 public _fee10000th = 0;
+    address payable public _feeCollector;
+
+    event FeeSetup(address feeCollector, uint256 fee10000th);
+
+    constructor(
+        AssetV2 asset,
```

Comment:
We have a problem here, this contract will be deployed on L1 where it's still using the Asset from 0.5 (not the V2).
So i'm suggesting to use the ERC1155 interface (we only use safeBatchTransferFrom I think) instead of AssetV2 class)

Reply:
done -> https://github.com/thesandboxgame/sandbox-smart-contracts/pull/617/commits/b43663be68f4c7623e0dce34330ed7f653b25e52